eth/tracers/api.go

```diff
@@ -337,10 +349,11 @@ func (api *API) traceChain(ctx context.Context, start, end *types.Block, config
 			}
 			// Prepare the statedb for tracing. Don't use the live database for
 			// tracing to avoid persisting state junks into the database.
-			statedb, err = api.backend.StateAtBlock(localctx, block, reexec, statedb, false)
-			if err != nil {
+			if _statedb, err := api.backend.StateAtBlock(localctx, block, reexec, statedb, false); err != nil {
```

Comment:
Perhaps rename the `_statedb` to `s`?

Reply:
Wait, is it necessary? if error is returned we will exit the for loop anyway. The statedb will not be used even it's set to nil.

Reply:
You're right, it isn't necessary. While investigating, I did use the statedb for outputting some more info, but that code is gone now, so I can revert that change